### DISCUSSION TITLE
Show apt updates

### DISCUSSION THREAD

**Discussion by kevincw01** [nicolargo/glances#2813] [Q&A] [answered]
Wondering if anyone has suggestions on how I would add a field in glances for whether the system has updates available via apt.

**Comment by nicolargo** [comment]
Yes, it is possible using an AMP: https://glances.readthedocs.io/en/latest/aoa/amps.html

Something like that in the Glances configuration file:

```
[amp_apt]
enable=true
refresh=300
one_line=true
command=<your script checking if update are available>
``` 

For the script, perhaps use "apt list --upgradable".

**Comment by nicolargo** [comment]
Or better for the script, use apt-check: 

```
/usr/lib/update-notifier/apt-check --human-readable --no-esm-messages 2>/dev/null
```

**Comment by nicolargo** [comment, marked as answer]
Test ok on my system:

![image](https://github.com/nicolargo/glances/assets/776747/a6e8b7a5-d70e-4615-9940-1a1dc2682a04)

Script in /tmp/foo.sh:

```
/usr/lib/update-notifier/apt-check --human-readable --no-esm-messages 2>/dev/null
```

Glances configuration file:

```
[amp_apt]
enable=true
refresh=300
one_line=true
command=/bin/sh /tmp/apt.sh
```

**Reply by kevincw01** [reply]
Thanks nico.  Gotta love the dell xps laptops too :)

**Reply by kevincw01** [reply]
@nicolargo is there a way to configure this as a "sensor" type instead as a new stand alone data type?  I ask because I'm comsuming this data from another api which doesnt support unique types and I'd prefer to not have to modify that API for this unique case.  Sensor type would be ideal for me.

**Reply by nicolargo** [reply]
What do you mean by "sensor" type ?

**Comment by kevincw01** [comment]
As in sensors category defined here among the others such as cpu and disk.  https://glances.readthedocs.io/en/latest/aoa/index.html

**Reply by nicolargo** [reply]
AMP (for Application Monitoring Process) is "sensor", it can be display in UI and retrieve through the API.

**Reply by kevincw01** [reply]
My apt category shows up a distinct type seperate from sensors, cpu, task, etc.  Maybe i am specifying it incorrectly?
![Screenshot_20240609_084919_Chrome](https://github.com/nicolargo/glances/assets/455203/d797f385-2a45-4adc-853a-af30f29eb4c5)

**Comment by basrieter** [comment]
Nice solution, however, if you run this in a Docker container to monitor the host, it will not work. Any suggestions for that?
